src/cpu/core_dyn_x86/risc_x64.h

```diff
@@ -439,7 +439,8 @@ static void gen_synchreg(DynReg * dnew,DynReg * dsynch) {
 	if ((dnew->flags ^ dsynch->flags) & DYNFLG_CHANGED) {
 		/* Ensure the changed value gets saved */	
 		if (dnew->flags & DYNFLG_CHANGED) {
-			dnew->genreg->Save();
+			if (GCC_LIKELY(dnew->genreg != NULL))
+				dnew->genreg->Save();
```

Comment:
Comment only - this is a nice optimization (in performance-critical code).
Looks like autotools takes care of macro'ing GCC_LIKELY so we can safely use it.